README.md

```diff
@@ -1,2 +1,314 @@
+
+## The Importance of Strict Compliance
+
+Strict compliance with the database schema offers numerous advantages:
+
+1. **Seamless ORM Integration:** These compliant structures are optimized for integration with Object-Relational Mapping (ORM) systems. ORM systems rely on the structure of these objects to perform various operations like data retrieval, updates, and deletions.
+
+2. **Reduced Complexity:** By adhering strictly to the database schema, you simplify the interaction between your Go application and the database. There's no need to handle additional fields that are not part of the schema.
+
+## Creating Structures
+
+For each table in your database schema, create a corresponding structure in your Go code. Ensure that the fields in these structures mirror the columns in the respective tables. Additionally, you can use struct tags to provide metadata about the fields, such as indicating the primary key.
+
+## Extending or Adding Keys
+
+If you need to extend the functionality or add extra keys to a structure, create a new structure to accommodate these changes. This new structure should embed the original database schema-compliant structure. In this way, you maintain separation between the strict schema structure and the extended version.
+
+
+## FindFirst Function
+
+The `FindFirst` function retrieves the first record from a specified database table based on the primary key field of a provided structure. This function is particularly useful when you want to retrieve the row with the smallest primary key value from a table.
+
+### Parameters:
+
+- `tableName` (string): The name of the database table from which to retrieve the record.
+- `structure` (interface{}): A pointer to a struct (e.g., &MyStruct{}) representing the structure into which the retrieved data will be scanned. The primary key field of the struct is used to determine the record to retrieve.
+
+### Return Value:
+
+- `error`: An error is returned if any of the following conditions are met:
+  - The provided structure is not a pointer to a struct.
+  - The primary key field is not found in the struct or is missing the "primarykey" tag.
+  - There is an issue with executing the SQL query or scanning the result.
+
+### Functionality:
+
+- The `FindFirst` function queries the specified database table to retrieve the first record based on the primary key field of the provided structure.
+
+- It checks if the `structure` argument is a pointer to a struct. If not, it returns an error.
+
+- It identifies the primary key field in the provided struct using the "primarykey" tag.
+
+- It constructs an SQL query to retrieve the first record from the specified `tableName` based on the primary key.
+
+- The query results are scanned into the provided `structure`, which should be a pointer to a struct.
+
+### Example Usage:
+
+```go
+var result MyStruct
+err := FindFirst("your_table", &result)
```

Comment:
where do we provide the primary key here ?, shouldn't it be your_table.FindFirst(id) ?

Reply:
we are getting the primarykey from the struct feilds  so that no argument will be provided.